tests/core_tests.py

```diff
@@ -551,6 +551,14 @@ def test_template_kwarg(self):
         rendered = tp.process_template(s, foo="bar")
         self.assertEqual("bar", rendered)
 
+    def test_custom_process_template(self):
```

Comment:
let's also have a test in the sqllab to show how query will be modified

Reply:
do you mean doing `run_sql` test?